lib/Page.js

```diff
@@ -98,6 +100,16 @@ class Page extends EventEmitter {
     client.on('Performance.metrics', event => this._emitMetrics(event));
   }
 
+  /**
+   * @param {!string} eventName
+   * @param {!Object} event
+   */
+  _handleRequest(eventName, event) {
```

Comment:
let's pass FrameManager over to the NetworkManager. This will make it easier to attribute requests with frames.

So in page constructor you'll have:

```js
this._frameManager = new FrameManager(client, frameTree, this);
this._networkManager = new NetworkManager(client, this._frameManager);  
```

and this code will move to the NetworkManager.

Reply:
https://github.com/GoogleChrome/puppeteer/pull/1646/commits/da9a0b06f984c0c219a04f21e280c8056085604d